### PR TITLE
use euid

### PR DESCRIPTION
**Pull request by raphaelgavache**
Remove cap_setuid, require root and use SetEuid instead

### CODE REVIEW

**Review by BaptisteFoy — APPROVED** [review]

File: omnibus/package-scripts/installer-rpm/posttrans [hunk 1]
```diff
@@ -65,9 +65,8 @@ fi
 
 # Set helper rights
 chmod 750 ${HELPER}
-setcap cap_setuid+ep ${HELPER}
 
-$BOOTSTRAP_INSTALLER bootstrap --url "oci://public.ecr.aws/datadog/installer-package:7.54.0-devel.git.366.99b427c.pipeline.32143438-1"
+$BOOTSTRAP_INSTALLER bootstrap --package "datadog-installer"
```
Comment (chouquette) [inline omnibus/package-scripts/installer-rpm/posttrans:69]: Could you clarify what this change is doing/why it's done?

File: pkg/installer/service/helper/main.go [hunk 2]
```diff
@@ -254,11 +177,29 @@ func executeCommand() error {
...
 		os.Exit(1)
 	}
 	os.Exit(0)
 }
+
+func setupPriviledges() error {
+	if os.Getuid() != 0 {
+		return fmt.Errorf("only root can execute this command")
+	}
+	if err := syscall.Setuid(os.Getuid()); err != nil {
```
Comment (chouquette) [inline pkg/installer/service/helper/main.go:198]: Is this required? If I understand this correctly you already checked that UID is 0, and this seems to be setting it to 0 again (I might not understand seteuid enough)

**Review by chouquette — APPROVED** [review]

File: omnibus/package-scripts/installer-rpm/posttrans (hunk 1 above)
Comment (raphaelgavache) [reply omnibus/package-scripts/installer-rpm/posttrans:69]: it allows to read from config which container registry to hit, and defaults to latest released version

File: pkg/installer/service/helper/main.go (hunk 2 above)
Comment (raphaelgavache) [reply pkg/installer/service/helper/main.go:198]: yes, it's an unfortunate hack required for subprocesses

**Review by chouquette — APPROVED** [review]

**Pull request merged**